src/Carbon/Carbon.php

```diff
@@ -396,11 +396,9 @@ public function __construct($time = null, $tz = null)
             )
         ) {
             $dateTime = new DateTime('now', $timezone);
-            $microTime = microtime(true) * 1000000 % 1000000;
-            if ($microTime > 0) {
-                $microTime = str_pad(strval($microTime), 6, '0', STR_PAD_LEFT);
-                $time = $dateTime->format(static::DEFAULT_TO_STRING_FORMAT).'.'.$microTime;
-            }
+            $microTime = microtime(true);
+            $microTime = str_pad(strval(($microTime - (int) $microTime) * 1000000 % 1000000), 6, '0', STR_PAD_LEFT);
```

Comment:
Can we just replace this 2 lines with code bellow?

```php
$microTime = substr(microtime(), 2, 6);
```

Reply:
How could this prevent from having negative values? `-1.25 - -1 = -0.25`, still negative

Reply:
or

```php
$microTime = microtime() * 1e6;
```